odoo/addons/base/models/res_partner.py

```diff
@@ -16,6 +17,10 @@
 from odoo import api, fields, models, tools, SUPERUSER_ID, _, Command
 from odoo.exceptions import RedirectWarning, UserError, ValidationError
 
+import typing
+if typing.TYPE_CHECKING:
+    from typing_extensions import Self
```

Comment:
if I do this, VScode doesn't understand typing anymore. It writes "any | Self' instead of 'Self'

Reply:
Ok, I think it is because you have python 3.10. What about the following (since `typing_extensions` is not installed after 3.10).

```python
try:
  from typing_extensions import Self
except ImportError:
  from typing import Self
```